server/settings_test.go

```diff
@@ -618,20 +654,65 @@ groups:
 
 					res, err := serverClient.Default.Server.ChangeSettings(&server.ChangeSettingsParams{
 						Body: server.ChangeSettingsBody{
-							AlertManagerURL:   url,
+							AlertManagerURL:   alertManagerURL,
 							AlertManagerRules: rules,
 						},
 						Context: pmmapitests.Context,
 					})
 					require.NoError(t, err)
-					assert.Equal(t, url, res.Payload.Settings.AlertManagerURL)
+					assert.Equal(t, alertManagerURL, res.Payload.Settings.AlertManagerURL)
 					assert.Equal(t, rules, res.Payload.Settings.AlertManagerRules)
 
 					gets, err := serverClient.Default.Server.GetSettings(nil)
 					require.NoError(t, err)
-					assert.Equal(t, url, gets.Payload.Settings.AlertManagerURL)
+					assert.Equal(t, alertManagerURL, gets.Payload.Settings.AlertManagerURL)
 					assert.Equal(t, rules, gets.Payload.Settings.AlertManagerRules)
 
+					t.Run("PrometheusReloaded", func(t *testing.T) {
+						time.Sleep(3 * time.Second) // Prometheus needs some time to reload configs.
```

Comment:
I think we should avoid sleep-driven development where possible, and there it is possible: https://prometheus.io/docs/prometheus/latest/management_api/#readiness-check

Reply:
checked both endpoints and both of them are returning 200 status code even if config is not reloaded yet.